src/Time/TimeStamp.hs

```diff
@@ -49,9 +74,75 @@ timeMul :: forall (unit :: Rat) . KnownRat unit
         -> Time unit
 timeMul n (Time t) = Time (n * t)
 
+{- | Operator version of 'timeMul'.
+
+>>> 3 *:* sec 5
+15s
+
+-}
+(*:*) :: forall (unit :: Rat) . KnownRat unit
+      => RatioNat -> Time unit -> Time unit
+(*:*) = timeMul
+
 -- | Returns the result of division of two 'Time' elements.
 timeDiv :: forall (unit :: Rat) . KnownRat unit
         => Time unit
         -> Time unit
         -> RatioNat
 timeDiv (Time t1) (Time t2) = t1 / t2
+
+{- | Operator version of 'timeDiv'.
+
+>>> sec 15 /:/ sec 3
+5 % 1
+
+-}
+(/:/) :: forall (unit :: Rat) . KnownRat unit
+      => Time unit -> Time unit -> RatioNat
+(/:/) = timeDiv
+
+-- | Sums times of different units.
+--
+-- >>> minute 1 +:+ sec 1
+-- 61s
+--
+(+:+) :: forall (unitResult :: Rat) (unitLeft :: Rat) . KnownDivRat unitLeft unitResult
+      => Time unitLeft
+      -> Time unitResult
+      -> Time unitResult
+t1 +:+ t2 = toUnit t1 + t2
+{-# INLINE (+:+) #-}
+
+-- | Substracts times of different units.
+--
+-- >>> minute 1 -:- sec 1
+-- 59s
+--
+(-:-) :: forall (unitResult :: Rat) (unitLeft :: Rat) . KnownDivRat unitLeft unitResult
+      => Time unitLeft
+      -> Time unitResult
+      -> Time unitResult
+t1 -:- t2 = toUnit t1 - t2
+{-# INLINE (-:-) #-}
+
+{- | Similar to '-:-' but more safe and returns order in pair with resulting time difference.
+
+>>> sec 5 -%- sec 3
+(GT,2s)
+
+>>> sec 5 -%- sec 6
+(LT,1s)
+
+-}
+(-%-) :: forall (unitResult :: Rat) (unitLeft :: Rat) . KnownDivRat unitLeft unitResult
+      => Time unitLeft
+      -> Time unitResult
+      -> (Ordering, Time unitResult)
+t1 -%- (Time t2Rat) =
+    let (Time t1Rat) = toUnit @unitResult t1
+        order = compare t1Rat t2Rat
+        newTime = Time $ case order of
+            LT -> t2Rat - t1Rat
+            GT -> t1Rat - t2Rat
+            EQ -> 0
+    in (order, newTime)
```

Comment:
I think you can avoid code duplication with `timeDiff` function if you implement function which takes two rational numbers and subtract them with returning `Ordering`.